scripts/jest/setupTests.js

```diff
@@ -13,8 +13,21 @@ if (process.env.REACT_CLASS_EQUIVALENCE_TEST) {
   // https://github.com/facebook/jest/blob/v20.0.4/packages/jest-matchers/src/spyMatchers.js#L160
   const isSpy = spy => spy.calls && typeof spy.calls.count === 'function';
 
-  const spyOn = global.spyOn;
+  // This is the Jasmine spyOn() method.
+  // Unlike jest.spyOn(), it does not pass through to the method it spies on by default.
+  const jasmineSpyOn = global.spyOn;
+
   const noop = function() {};
+  const spyOn = function(object, methodName) {
+    if (object === console) {
```

Comment:
Note to self: Some of our tests spy on `console.log` (eg [`ReactDOMComponent-test`](https://github.com/facebook/react/blob/b5334a44e99c32ca26f7d65ca8312e13f7b7f4b7/packages/react-dom/src/__tests__/ReactDOMComponent-test.js#L1369)) so this would also need to check if `methodName === "warn" || methodName === "error"`

Reply:
We could also change them to use `jest.fn` or something directly.